synfig-studio/src/gui/widgets/widget_timetrack.cpp

```diff
@@ -198,6 +198,44 @@ bool Widget_Timetrack::move_selected(synfig::Time delta_time)
 	return ok;
 }
 
+ void Widget_Timetrack::interpolate_selected(synfig::Interpolation type)
+{
+	synfig::Waypoint::Model model;
+	model.set_before(type);
+	model.set_after(type);
+	std::vector<WaypointItem*> selection = waypoint_sd.get_selected_items(); //storing selected waypoint items
+
+	for(int i =0 ; i<selection.size() ; i++) //iteratre through selected waypoint items to fetch the set and apply the interpolation type
```

Comment:
1. You don't need the comments here. Neither the ones on previous line.
2. Prefer [Range-based for loop](https://www.geeksforgeeks.org/range-based-loop-c/).

Reply:
done